components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/AuthenticationResponseProcessor.java

```diff
@@ -0,0 +1,244 @@
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    @Override
+    public ActionType getSupportedActionType() {
+
+        return ActionType.AUTHENTICATION;
+    }
+
+    @Override
+    public ActionExecutionStatus processSuccessResponse(Map<String, Object> eventContext, Event event,
+                                                        ActionInvocationSuccessResponse actionInvocationSuccessResponse)
+            throws ActionExecutionResponseProcessorException {
+
+        AuthenticationContext context = (AuthenticationContext) eventContext.get(
+                AuthenticatorAdapterConstants.AUTH_CONTEXT);
+        ActionInvocationResponse.Status actionStatus = actionInvocationSuccessResponse.getActionStatus();
+
+        if (ActionInvocationResponse.Status.SUCCESS.equals(actionStatus)) {
+            List<PerformableOperation> operationsToPerform = actionInvocationSuccessResponse.getOperations();
+            try {
+                AuthenticationRequestUser authRequestUser = new AuthenticationRequestUser(StringUtils.EMPTY);
+                UserStore userStore = new UserStore(StringUtils.EMPTY);
+                if (operationsToPerform != null) {
+                    for (PerformableOperation operation : operationsToPerform) {
+                        performOperation(operation, authRequestUser, userStore);
+                    }
+                }
+                context.setSubject(createAuthenticateduser(authRequestUser, context, userStore));
+            } catch (UserStoreException e) {
+                throw new ActionExecutionResponseProcessorException("Error occurred when trying to build authenticated " +
+                        "user from the external authenticator service response." ,e);
+            }
+        }
+
+        // TODO: Log operation process results.
+        return new ActionExecutionStatus(ActionExecutionStatus.Status.SUCCESS, eventContext);
+    }
+
+    @Override
+    public ActionExecutionStatus processErrorResponse(Map<String, Object> eventContext,
+                                                      Event actionEvent,
+                                                      ActionInvocationErrorResponse errorResponse) throws
+            ActionExecutionResponseProcessorException {
+
+        return new ActionExecutionStatus(ActionExecutionStatus.Status.ERROR, eventContext);
+    }
+
+    /**
+     * Validate the given performableOperation and apply it to the user attribute.
+     *
+     * @param performableOperation  PerformableOperation from the external authentication service response.
+     */
+    public void performOperation(PerformableOperation performableOperation, AuthenticationRequestUser authRequestUser,
+                                 UserStore userStoreDomain) throws ActionExecutionResponseProcessorException {
+
+        if (!isAllowedOperation(performableOperation.getOp())) {
+            return;
+        }
```

Comment:
Based on the allowed operations defined in the request, this validation should have happened in the action core execution and is redundant here

Reply:
Addressed with commit 81da7fbd3e70fa2c10140203df128edc4574c5a1